src/main/java/com/rackspace/ceres/app/services/MetadataService.java

```diff
@@ -127,75 +107,57 @@ public MetadataService(ReactiveCqlTemplate cqlTemplate,
 
   public Publisher<?> storeMetadata(String tenant, String seriesSetHash, Metric metric) {
     final CompletableFuture<Boolean> result = seriesSetExistenceCache.get(
-            new SeriesSetCacheKey(tenant, seriesSetHash),
-            (key, executor) ->
-                    cassandraTemplate.exists(
-                                    query(
-                                            where(COL_TENANT).is(tenant),
-                                            where(COL_SERIES_SET_HASH).is(seriesSetHash)
-                                    ),
-                                    SeriesSetHash.class
-                            )
-                            .name("metadataExists")
-                            .metrics()
-                            .retryWhen(appProperties.getRetryInsertMetadata().build())
-                            .doOnError(t -> log.error("Exception in exists check "+t.getMessage()))
-                            .doOnNext(exists -> {
-                              if (exists) {
-                                cassandraHit.increment();
-                              } else {
-                                cassandraMiss.increment();
-                              }
-                            })
-                            // not cached in cassandra?
-                            .flatMap(exists -> exists ?
-                                    Mono.just(true) :
-                                    // not cached, so store the metadata to be sure
-                                    storeMetadataInCassandra(tenant, seriesSetHash, metric)
-                                        .and(elasticSearchService.saveMetricToES(tenant, metric))
-                                            .map(it -> true))
-                            .toFuture()
+        new SeriesSetCacheKey(tenant, seriesSetHash),
+        (key, executor) ->
+            cassandraTemplate.exists(
+                    query(
+                        where(COL_TENANT).is(tenant),
+                        where(COL_SERIES_SET_HASH).is(seriesSetHash)
+                    ),
+                    SeriesSetHash.class
+                )
+                .name("metadataExists")
+                .metrics()
+                .retryWhen(appProperties.getRetryInsertMetadata().build())
+                .doOnError(t -> log.error("Exception in exists check " + t.getMessage()))
+                .doOnNext(exists -> {
+                  if (exists) {
+                    cassandraHit.increment();
+                  } else {
+                    cassandraMiss.increment();
+                  }
+                })
+                // not cached in cassandra?
+                .flatMap(exists -> exists ?
+                    Mono.just(true) :
+                    // not cached, so store the metadata to be sure
+                    storeMetadataInCassandra(tenant, seriesSetHash, metric)
```

Comment:
why are we still storing in cassandra?

Reply:
SeriesSetHash and SeriesSet tables are going to be as it were. these are being used by downsampling process and metrics deletion classes. If we delete these, we will have to query ES for them and I think idea of using ES are surely for storing metadata but avoiding calls to ES from ceres for its internal usage. In that case, we won't be taking headache of impact of ES failure on ceres system, will be effecting only metadata apis.